src/main/java/edu/kis/powp/jobs2d/command/visitor/VisitorRotate.java

```diff
@@ -0,0 +1,53 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+import java.util.Iterator;
+
+public class VisitorRotate implements VisitorCommand{
+	private final double cos;
+	private final double sin;
+
+	public VisitorRotate(double angle) {
+		this.cos = Math.cos(Math.toRadians(angle));
+		this.sin = Math.sin(Math.toRadians(angle));
+	}
+
+	@Override
+	public void visit(OperateToCommand operateToCommand) {
+		int x = operateToCommand.getPosX();
+		int y = operateToCommand.getPosY();
+		operateToCommand.setPosX(this.calculateNewPositionX(x, y));
+		operateToCommand.setPosY(this.calculateNewPositionY(x, y));
```

Comment:
this.calculateNewPositionY(x, y) method approach could be applied to other transformations.

Reply:
refactored - added an additional abstraction layer